Objects/unicodeobject.c

```diff
@@ -16070,6 +16070,58 @@ _PyUnicode_Fini(PyInterpreterState *interp)
     _PyUnicode_FiniEncodings(&state->fs_codec);
 
     unicode_clear_identifiers(state);
+
+    // Clear the single character singletons
+    for (int i = 0; i < 128; i++) {
+        PyASCIIObject *ascii = (PyASCIIObject *)& _Py_SINGLETON(strings).ascii[i];
+        if (ascii->wstr != NULL) {
+            PyObject_FREE(ascii->wstr);
+            ascii->wstr = NULL;
+        }
+    }
+    for (int i = 0; i < 128; i++) {
+        PyASCIIObject *ascii = (PyASCIIObject *)&_Py_SINGLETON(strings).latin1[i];
+        PyCompactUnicodeObject *compact = (PyCompactUnicodeObject *)ascii;
+        if (ascii->wstr != NULL) {
+            PyObject_FREE(ascii->wstr);
+            ascii->wstr = NULL;
+            compact->wstr_length = 0;
+        }
+        if (compact->utf8 != NULL) {
+            PyObject_Free(compact->utf8);
+            compact->utf8 = NULL;
+            compact->utf8_length = 0;
+        }
+    }
+}
+
+
+void
+_PyStaticUnicode_Dealloc(PyObject *op)
```

Comment:
Would it make sense to share code between this function and unicode_dealloc()? Maybe add an unicode_dealloc_common() function?

Reply:
Yes, there is the duplication of the `PyObect_Free()` calls, but not really to the level of what I would consider "in common".  I did, however, refactor out the duplication of the 1-character singletons and frozen strings deallocation into a standalone static function.